# 161342833 configure coveralls to check code quality

wiztemple opened the pull request:
### What does this PR do?
configure the application to report code coverage

### Description of task to be completed
- install coverall, Istanbul, and nyc
- obtain a repo token for the .coveralls.yml file
- set up a script to run code coverage


### How should this be manually tested?
- clone the repo and cd into it ch-integrate-coveralls-code-161342833
- run _npm run coverage_

### What are the relevant Pivotal Tracker stories?
#161342833

PriscillaSam commented:
Hi @wiztemple.
There's no need for the extra lines after the relevant PT story on the pull request.

ucheg6 reviewed (commented):
@wiztemple I don't think the '.nyc_output' file should be on github. My observation though. You can gitignore it

seyi-adeleke commented on server/example/example.js:
```diff
@@ -0,0 +1,4 @@
+const arr = [1, 2, 3];
```
we already have tests passing on develop. You shouldn't need to do this

seyi-adeleke commented on server/tests/sample.test.js:
```diff
@@ -1,9 +1,12 @@
-    it('should return -1 when the value is not present', () => {
-      equal([1, 2, 3].indexOf(4), -1);
-    });
+  it('testing array', () => {
```
Leave this as is

seyi-adeleke commented on package.json:
```diff
@@ -8,10 +8,11 @@
     "build": "npm run clean && mkdir dist/ && npm run babel-server",
     "clean": "rm -rf dist/",
-    "test": "mocha --require babel-register server/tests/*.test.js",
+    "test": "mocha --require babel-core/register ./server/tests/*.test.js  --exit",
```
whats happening here?

wiztemple replied:
noted. Thanks

seyi-adeleke replied:
What are you noting please? I'd like a response to the question. To be specific why do we need the `--exit` flag and the `babel-core` change

wiztemple replied:
the --exit flag is to end/exit the test after running. the babel-core change was a mistake. from the initial command on the script "mocha --require babel-core/register", the command is to pick either pick babel-core or babel-register depending on which one is available. so I checked my package.json and picked babel-register because it was in the package.json.

seyi-adeleke commented on .coveralls.yml:
```diff
@@ -0,0 +1 @@
+ repo_token: cnwBfq9loEWhbObghcFjjnl3IcB6d99Su
```
🤔 This should probably be kept in a .env file.

wiztemple replied:
noted. thanks

seyi-adeleke commented:
Also your PR title is wrong

wiztemple commented:
> Also your PR title is wrong

Noted, thanks.

Commit:
chore(coveralls): configure coveralls for code coverage

- install coveralls, istanbul and nyc

- add .coveralls.yml file and include repo token

- write a script to run configuration
